extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -826,6 +835,11 @@ public static JsonArray getJsonResponse(@Nonnull final Page page,
                         .value("hl", localization.getLocalizationCode())
                         .value("gl", contentCountry.getCountryCode())
                     .end()
+                    .object("user")
+                        // TO DO: provide a way to enable restricted mode with:
+                        // .value("enableSafetyMode", boolean)
```

Comment:
Currently it is implemented in NewPipe using a cookie: `PREF=f2=8000000, in the `DownloaderImpl` class.

https://github.com/TeamNewPipe/NewPipe/blob/c398308872f617829352ed283a464c0f105ad8ef/app/src/main/java/org/schabi/newpipe/DownloaderImpl.java#L162

I agree it should be moved to the extractor, using that boolean. But does it still work?

Reply:
I don't know, it needs testing.